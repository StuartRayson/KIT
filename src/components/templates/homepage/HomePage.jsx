import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ContentApiSrvc from '../../../shared/services/contentApiSrvc';
import Header from '../../organisms/header/Header.jsx';
import Article from '../../molecules/article/Article.jsx';

class HomePage extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      articles: [],
      loading: true,
      error: null
    };
  }

  componentDidMount() {
    ContentApiSrvc.getArticles()
      .then(data => this.setState({
        articles: data,
        loading: false,
        error: null
      }))
      .catch((reason) => {
        this.setState({
          error: reason,
          loading: false
        });
      });
  }

  render() {
    const { loading, error, articles } = this.state;
    const { errorMessage } = this.props;

    return (
      <div className="t-homepage">
        <Header className="t-homepage__header" />

        <div>
          {loading && (
            <div style={{ textAlign: 'center', fontSize: '42px' }}>
              <span role="img" aria-label="loading">
                âŒ›
              </span>
            </div>
          )}
          {error && <div style={{ textAlign: 'center', fontSize: '42px' }}>{errorMessage}</div>}

          {articles.map(article => (
            <Article key={article.id} title={article.title} content={article.body} />
          ))}
        </div>
      </div>
    );
  }
}

HomePage.propTypes = {
  errorMessage: PropTypes.string
};

export default HomePage;
