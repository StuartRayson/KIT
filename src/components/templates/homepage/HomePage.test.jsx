import React from 'react';
import { shallow } from 'enzyme';
import HomePage from './HomePage.jsx';

const mockArticles = [
  {
    id: 0,
    title: 'title1',
    body: 'body1'
  },
  {
    id: 1,
    title: 'title2',
    body: 'body2'
  }
];

jest.mock('../../../shared/services/contentApiSrvc.js', () => ({
  getArticles: jest.fn(() => ({
    then: (cb) => {
      cb(mockArticles);
      return {
        catch: jest.fn()
      };
    }
  }))
}));

describe('<HomePage />', () => {
  it('renders without crashing', () => {
    shallow(<HomePage />);
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(<HomePage />);

    expect(wrapper).toMatchSnapshot();
  });
});
