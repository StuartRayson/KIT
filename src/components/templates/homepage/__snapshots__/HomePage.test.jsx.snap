// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match snapshot with basic config 1`] = `
<div
  className="t-homepage"
>
  <Header
    className="t-homepage__header"
  />
  <div>
    <Article
      content="body1"
      key="0"
      title="title1"
    />
    <Article
      content="body2"
      key="1"
      title="title2"
    />
  </div>
</div>
`;
